capstone-backend/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -45,29 +45,42 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    // response.setHeader("Access-Control-Allow-Methods", "GET");
-    // response.setHeader("Access-Control-Allow-Credentials", "true");
-    // response.setHeader("Access-Control-Allow-Origin",
-    //     "https://3000-bbaec244-5a54-4467-aed6-91c386e88c1a.ws-us02.gitpod.io");
-
     String fullOutput = "";
-    String searchTerm = request.getParameter("searchTerm");
-    if (searchTerm == null || searchTerm.isEmpty())
-      return;
-    searchTerm = URLEncoder.encode(searchTerm, "UTF-8");
+    String formattedURL = "";
 
-    int maxResults = 0;
-    String maxResultsParam = request.getParameter("maxResults"); 
-    if (maxResultsParam != null) { 
-      maxResults = parseNaturalNumber(maxResultsParam);
-    }
-    if (maxResultsParam == null || maxResults == -1) {
-      maxResults = DEFAULT_MAX_RESULTS;
+    if (request.getParameter("gbookId") != null) {
+      if (request.getParameterMap().size() > 1) {
+        System.err.println("Error: No other parameter can be sent with a gbookId");
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      // make formatted url
+      String gbookId = request.getParameter("gbookId");
+      formattedURL = String.format("https://www.googleapis.com/books/v1/volumes/%s", gbookId);
+
+    } else if (request.getParameter("searchTerm") != null || !request.getParameter("searchTerm").isEmpty()) {
```

Comment:
Should there also be a parameterMap size check inside this else-if clause? This would make sure nothing else is sent along with searchTerm.

Reply:
Yeah that's a good idea! I've now incorporated it using what you said when we discussed offline to check for 2 or more parameters, then confirm that maxResults is not null.